src/psyclone/tests/psyir/nodes/acc_directives_test.py

```diff
@@ -495,3 +496,83 @@ def test_accdatadirective_update_data_movement_clauses(fortran_reader,
     # 'sfactor' should have been removed from the copyin()
     assert ("!$acc data copyin(small_holding,small_holding(3)%grid,"
             "small_holding(3)%grid(jf)%data), copy(sto_tmp)" in output)
+
+
+def test_acc_atomics_is_valid_atomic_statement(fortran_reader):
+    ''' Test the ACCAtomicDirective can identify when a statement is a valid
+    expression to support OpenACC atomics. '''
+
+    code = '''
+    subroutine my_subroutine()
+        integer, dimension(10, 10) :: A = 1
+        integer, dimension(10, 10) :: B = 2
+        integer :: i, j, val
+
+        A(1,1) = A(1,1) * 2
+        A(1,1) = A(1,1) / 2 + 3 - 5
+        A(1,1) = MAX(A(1,1), A(1,2))
+    end subroutine
+    '''
+    tree = fortran_reader.psyir_from_source(code)
+    for stmt in tree.walk(Assignment):
+        assert ACCAtomicDirective.is_valid_atomic_statement(stmt)
+
+    code = '''
+    subroutine my_subroutine()
+        integer, dimension(10, 10) :: A = 1
+        integer, dimension(10, 10) :: B = 2
+        integer :: i, j, val
+
+        A(1,1) = A(1,1) ** 2
+        A(:,1) = A(:,1) / 2 + 3 - 5
+        A(1,1) = MOD(A(1,1), 3)
```

Comment:
How come we exclude MOD? Is there some particular reason?

Reply:
It is not in the list of supported atomic instrinsics :) And I tried with the compiler just in case and they certainly don't like it.

Reply:
Aha!